opencensus/trace/internal/b3.cc

```diff
@@ -75,17 +75,19 @@ SpanContext FromB3Headers(absl::string_view b3_trace_id,
   std::string trace_id_binary = absl::HexStringToBytes(b3_trace_id);
   std::string span_id_binary = absl::HexStringToBytes(b3_span_id);
 
-  // Extend 64-bit trace_id to 128-bit.
+  // trace_id_ptr must point to a 128-bit trace_id.
+  const uint8_t* trace_id_ptr =
+      reinterpret_cast<const uint8_t*>(trace_id_binary.data());
   uint8_t extended_trace_id[16];
-  if (b3_trace_id.length() == 16) {
+  if (trace_id_binary.length() != 16) {
```

Comment:
Switching to != 16 is the better option, but now you have an issue because there is zero guarantee that != 16 means size=8. Please make this code resilient to changing sizes.

Reply:
L65 guarantees it'll be either 16 or 8 after HexStringToBytes. I'll try duplicating the check here and let's see if it'll be more readable.